recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -67,9 +67,17 @@ def config_options(self):
         if self.settings.os == "Windows":
             self.options.rm_safe("fPIC")
             del self.options.with_etw
+            # Opentelemetry-cpp on Windows only supports static libraries,
+            # due to ppor support from upstream (and build failures)
+            # As Upstream does not expect shared builds to work properly on Windows,
+            # instead of validating shared out (which would remove shared builds from CCI of dependants),
+            # we just force static builds on Windows.
+            # See https://github.com/open-telemetry/opentelemetry-cpp/issues/2477
+            self.package_type = "static-library"
```

Comment:
The title of the PR "add shared support for all platforms" is confusing when the PR is just refactoring how windows is handled, where `shared` remains unsupported


I would personally also remove the comment about "so that builds from CCI".. we are making a lot of efforts such that recipes are agnostic and no longer get logic in them that is exclusive to our CI builds. Arguably, I don't think this change is exclusively motivated by our CI at all, but a user experience change - have the recipe fallback on Windows to the one configuration that actually works, as opposed to error out when `*:shared=True`, as that would match the behaviour outside of Conan too.